### PR TITLE
Filtering fix

### PR DESCRIPTION
**Pull request by msargentibm**
Filtering logic was happening, but the resulting object wasn't being passed through to the final  evaluate. I moved the logic to the final evaluate statement.

I'm not a huge fan of the extra method I added, but it made it easy to write tests quickly.

There was also a NullPointerException for the CodeKey class if System was null and equals was called. I added a test and a code fix.

### CODE REVIEW

**Review by csandersdev — APPROVED** [review]
LGTM. Thanks for doing this.

**Comment by msargentibm** [comment]
I should note that I've only briefly inspected the output from a test run to make sure that measures were being reported in their expected contexts. I didn't do a deep dive into any individual results.

**Review by jaigoradia — APPROVED** [review]
Approved

File: cohort-evaluator-spark/src/test/java/com/ibm/cohort/cql/spark/SparkCqlEvaluatorTest.java [hunk 1]
```diff
@@ -136,6 +139,107 @@ public void testAllTypesEvaluationSuccess() throws Exception {
...
+                makeEvaluationRequest("A"),
+                makeEvaluationRequest("A"),
+                makeEvaluationRequest("B")
+        );
+
+        requests.setEvaluations(evaluations);
+
+        List<CqlEvaluationRequest> cqlEvaluationRequests = evaluator.filterRequests(requests, "A");
+
+        assertEquals(2, cqlEvaluationRequests.size());
```
Comment (dkwasny-ibm) [inline cohort-evaluator-spark/src/test/java/com/ibm/cohort/cql/spark/SparkCqlEvaluatorTest.java:200]: Pedantic: Not required, but some assertion to ensure the two elements in the list are the correct ones would make this test more robust.

**Review by dkwasny-ibm — APPROVED** [review]

**Pull request merged**